### ISSUE TITLE
Nullability of static field doesn't seem to be tracked

### ISSUE THREAD

**Issue by jcouv** [dotnet/roslyn#26651]
[Example](https://sharplab.io/#v2:EYLgZgpghgLgrgJwgZwLQDk4BstWFiAJQkiQDsBjCAFQE8AHFAGgBMQBqAHwFgAoAAQDMAAn4AmYQGFhfAN59hi0QEYAbMKRQWAezJZaKgAwB+YcmEBeYWWxYA3AqVDRAFmEBZABQBKGbyXC8v4BSgCWYMKe5gCEVjY43o4hgUnJAJDIAHTU2gDKMAihZADmPnbCAPQVwgDuUAhkRcVSuQAcqoZiIMIACtrIyKH4EMIsEEikEJQj2hFQ1rYaJONTVJmpAQC+qdu8m0A=)

```C#
public class C 
{
    static readonly string? s = null;
    public void M() 
    {
        if (s != null)
        {
         	s.ToString(); // warning CS8602: Possible dereference of a null reference.
        }
    }
}
```

Found in nullable dogfood.
Tagging @cston

**Comment by jcouv** [comment]
Here's another symptom, likely from the same cause:
```
public class C 
{
    static readonly string[] x = new string[] { null }; // no warning
}
```

**Comment by cston** [comment]
The warning is not reported in the second example because we're currently skipping `NullableWalker` for implicit members. If the initializer is moved to an explicit `.cctor`, the warning is reported. We should have a separate bug to handle implicit members such as the default `.ctor` and `.cctor`.

**Comment by cston** [comment]
Logged https://github.com/dotnet/roslyn/issues/31139 for the issue with implicit constructors.
